server/http/helloapp/oauth.go

```diff
@@ -0,0 +1,128 @@
+package helloapp
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/mattermost/mattermost-plugin-api/experimental/bot/logger"
+	"github.com/mattermost/mattermost-plugin-api/experimental/oauther"
+	"github.com/mattermost/mattermost-plugin-apps/server/apps"
+	"github.com/mattermost/mattermost-plugin-apps/server/constants"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+func (h *helloapp) InitOAuther() error {
+	oauth2Config, err := h.GetOAuthConfig()
+	if err != nil {
+		return err
+	}
+	h.OAuther = oauther.NewFromClient(h.apps.Mattermost,
+		*oauth2Config,
+		h.finishOAuth2Connect,
+		logger.NewNilLogger(),
+		oauther.OAuthURL(constants.HelloAppPath+PathOAuth2),
+		oauther.StorePrefix("hello_oauth_"))
+	return nil
+}
+
+func (h *helloapp) handleOAuth(w http.ResponseWriter, req *http.Request) {
+	if h.OAuther == nil {
+		http.Error(w, "OAuth not initialized", http.StatusInternalServerError)
+		return
+	}
+	h.OAuther.ServeHTTP(w, req)
+}
+
+func (h *helloapp) GetOAuthConfig() (*oauth2.Config, error) {
+	conf := h.apps.Configurator.GetConfig()
+
+	id, secret, err := h.getOAuth2AppCredentials()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve App OAuth2 credentials")
+	}
+
+	return &oauth2.Config{
+		ClientID:     id,
+		ClientSecret: secret,
+		Endpoint: oauth2.Endpoint{
+			AuthURL:  conf.MattermostSiteURL + "/oauth/authorize",
+			TokenURL: conf.MattermostSiteURL + "/oauth/access_token",
+		},
+		RedirectURL: h.AppURL(PathOAuth2Complete),
```

Comment:
RedirectURL is not needed. It is handled here: https://github.com/mattermost/mattermost-plugin-api/blob/8b14e62b421ca2816901165e9ff750895451e621/experimental/oauther/oauth2.go#L102